openapi-cli/src/main/java/io/ballerina/openapi/cmd/SubCmdBase.java

```diff
@@ -0,0 +1,331 @@
+
+    public abstract String getDefaultFileName();
+
+    public Optional<OpenAPI> getFlattenOpenAPI(OpenAPI openAPI) {
+        // Flatten the OpenAPI definition with `flattenComposedSchemas: true` and `camelCaseFlattenNaming: true`
+        InlineModelResolver inlineModelResolver = new InlineModelResolver(true, true);
+        inlineModelResolver.flatten(openAPI);
+        return Optional.of(openAPI);
+    }
+
+    public Optional<OpenAPI> getFilteredOpenAPI(String openAPIFileContent) {
+        // Read the contents of the file with default parser options
+        // Flattening will be done after filtering the operations
+        SwaggerParseResult parserResult = new OpenAPIParser().readContents(openAPIFileContent, null,
+                new ParseOptions());
+        if (!parserResult.getMessages().isEmpty() &&
+                parserResult.getMessages().contains(UNSUPPORTED_OPENAPI_VERSION_PARSER_MESSAGE)) {
+            errorStream.println(ERROR_UNSUPPORTED_OPENAPI_VERSION);
+            return Optional.empty();
+        }
+
+        OpenAPI openAPI = parserResult.getOpenAPI();
+        if (Objects.isNull(openAPI)) {
+            errorStream.println(ERROR_OCCURRED_WHILE_PARSING_THE_INPUT_OPENAPI_FILE);
+            if (!parserResult.getMessages().isEmpty()) {
+                errorStream.println(FOUND_PARSER_DIAGNOSTICS);
+                parserResult.getMessages().forEach(errorStream::println);
+            }
+            return Optional.empty();
+        }
+
+        filterOpenAPIOperations(openAPI);
+        return Optional.of(openAPI);
+    }
+
+    @Override
+    public String getName() {
+        return cmdType.getName();
+    }
+
+    @Override
+    public void printLongDesc(StringBuilder stringBuilder) {
+        //This is the long description of the command and all handle within help command
+    }
+
+    @Override
+    public void printUsage(StringBuilder stringBuilder) {
+        //This is the usage description of the command and all handle within help command
+    }
+
+    @Override
+    public void setParentCmdParser(CommandLine commandLine) {
+        //This is not used in this command
+    }
+
+    private void populateInputOptions() {
+        if (Objects.nonNull(format)) {
+            if (!format.equalsIgnoreCase(JSON) && !format.equalsIgnoreCase("yaml")) {
+                setDefaultFormat();
+                errorStream.println(WARNING_INVALID_OUTPUT_FORMAT);
+            }
+        } else {
+            setDefaultFormat();
+        }
+
+        if (Objects.isNull(fileName)) {
+            fileName = String.format(getDefaultFileName(), cmdType.getName());
+        }
+
+        if (Objects.nonNull(outputPath)) {
+            targetPath = Paths.get(outputPath).isAbsolute() ?
+                    Paths.get(outputPath) : Paths.get(targetPath.toString(), outputPath);
+        }
+    }
+
+    private void setDefaultFormat() {
+        format = inputPath.endsWith(JSON_EXTENSION) ? JSON : "yaml";
```

Comment:
Shall we use constant for `yaml`?

Reply:
Fixed via [ad24c75](https://github.com/ballerina-platform/openapi-tools/pull/1770/commits/ad24c758c60adc0ffb60cdb0461b955c5994e913)